xarray/tests/test_netcdftimeindex.py

```diff
@@ -40,12 +42,16 @@ def date_dict(year=None, month=None, day=None,
     ('1999-01-01T12:34:56.78', date_dict(
         year='1999', month='01', day='01', hour='12', minute='34',
         second='56.78'))
-], ids=['year', 'month', 'month-dash', 'day', 'day-dash', 'hour', 'hour-dash',
-        'minute', 'minute-dash', 'second', 'second-dash', 'second-dec',
-        'second-dec-dash'])
-def test_parse_iso8601(string, expected):
-    from xarray.core.netcdftimeindex import parse_iso8601
+]
+ISO8601_STRING_TEST_IDS = [
```

Comment:
My thinking here was to use something like a dict, just to logically join together test names and parameters in the same place, e.g.,
```
ISO8601_STRING_TESTS = {
    'year': ('1999', date_dict(year='1999')),
    'month': ('199901', date_dict(year='1999', month='01')),
    ...
}

@pytest.mark.parmetrize(('string', 'expected', ISO8601_STRING_TESTS.values(),
                        ids=ISO8601_STRING_TESTS.keys())
```